lib/util.dart

```diff
@@ -127,10 +127,30 @@ class Util {
   /// Utility method to render stuff on a specific place.
   ///
   /// Some render methods don't allow to pass a offset.
-  /// This method translate the canvas, draw what you want, and then translate back.
-  void drawWhere(Canvas c, Vector2 p, void Function(Canvas) fn) {
+  /// This method translate the canvas before rendering your block.
+  /// The changes are reset after the block is run.
+  void renderAt(Canvas c, Vector2 p, void Function(Canvas) block) {
+    c.save();
     c.translate(p.x, p.y);
-    fn(c);
-    c.translate(-p.x, -p.y);
+    block(c);
```

Comment:
What does `block` mean here? Like a code block? Maybe `renderFunction` could work?

Reply:
ah yes, it means lambda. some languages call it block

Reply:
Ah, never heard that before.

Reply:
Not sure I liked the name `block` found it to be a little confusing, I think I prefer `fn` or even `renderFn`. Anyway, just a nit, feel free to ignore

Reply:
I will go back to fn